subsys/bootloader/Kconfig

```diff
@@ -74,7 +74,8 @@ config PM_PARTITION_SIZE_B0_IMAGE
 	default 0x7800 if !B0_MIN_PARTITION_SIZE && (SOC_NRF5340_CPUNET)
 	default FPROTECT_BLOCK_SIZE if SOC_SERIES_NRF91X || SOC_NRF5340_CPUAPP
 	default 0x3800 if SOC_NRF5340_CPUNET
-	default 0x7800 if SOC_NRF54L15_CPUAPP
+	default 0x9800 if SOC_NRF54L15_CPUAPP && !FPROTECT
```

Comment:
Just for my own shake, why do we set a different size depending on the FPROTECT?

Reply:
If you disable FPROTECT you can fit more into b0, because fprotect can only protect 31k.
So, for example, if you want to enable a lot of stuff to debug, you may not fit in the FPROTECT area and need to disable it, then it automatically also brings the size of allowed code up.